widgetssdk/src/main/java/com/glia/widgets/core/operator/GliaOperatorMediaRepository.java

```diff
@@ -26,14 +26,13 @@ public void removeMediaStateListener(OperatorMediaStateListener listener) {
     }
 
     public void onEngagementStarted(Engagement engagement) {
-        engagement.getMedia()
-                .on(Media.Events.OPERATOR_STATE_UPDATE, operatorMediaStateConsumer);
+        engagement.getMedia().on(Media.Events.OPERATOR_STATE_UPDATE, operatorMediaStateConsumer);
     }
 
-    public void onEngagementEnded(Engagement engagement) {
+    public void onEngagementEnded() {
         currentMediaState = null;
         notifyOperatorMediaStateChanged(null);
-        engagement.getMedia().off(Media.Events.OPERATOR_STATE_UPDATE, operatorMediaStateConsumer);
+//        engagement.getMedia().off(Media.Events.OPERATOR_STATE_UPDATE, operatorMediaStateConsumer);
```

Comment:
Is it a leftover or you would like to remove it at all?

Reply:
I want to remove it at all. The result should be to have all the subscribe and unsubscribe events in one place(EngagementRepository) that will share states with the different UseCases